### PR TITLE
Governance guidelines

### PR DESCRIPTION
**Pull request by rcurtin**
As a part of our acceptance to NumFOCUS, we've been asked to formalize our governance procedures.  In the past I don't think we've found this necessary, but it certainly can't hurt.  In this draft I've tried to formalize the procedures that pretty closely reflect what we already do.

This is open for comments from anyone!  If you have any comments, please feel free to add them and we can discuss.

I'll merge this (in whatever form it ends up being) a week after discussion ends, to give time for anyone to comment.

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+mlpack aims to be an open and welcoming environment, and as such, we have a
+code of conduct that helps foster this environment.  See
+(here)[https://github.com/mlpack/mlpack/blob/master/CODE_OF_CONDUCT.md] for
+more information.
+
+## Teams & Roles
+
+To keep overhead minimal, mlpack's teams and roles are simple: there is only
+the [Committers](https://github.com/orgs/mlpack/teams/contributors) team, and
+the [NumFOCUS leadership team](TODO:link).
```
Comment (zoq) [inline GOVERNANCE.md]: Don't think NumFOCUS provides such information, but I guess a simple list does work as well.

Comment (rcurtin) [reply GOVERNANCE.md]: Oh, I was just going to use Github teams to create a NumFOCUS leadership team.  But I was going to wait until we finalized this to create the team, then update the link here right before merge.

Comment (zoq) [reply GOVERNANCE.md]: Github teams is a good idea.

**Comment by rcurtin** [comment]
Seems like this was not controversial. :) I will go ahead and merge it. We can always update it later if we need.

**Pull request merged**